### PR TITLE
Add quests for `capacity=`, `capacity:disabled=` and `orientation=` for `amenity=parking`

### PR DESCRIPTION
**Pull request by wielandb**
This PR adds 3 quests relating to parking mapped as `amenity=parking`:

- How many cars can park here?
- How many parking spots for disabled people are there? 
- In what orientation do cars park here?

|   |   |
|---|---|
|photo_2023-11-21_23-50-13| photo_2023-11-21_23-51-04 | 
| photo_2023-11-21_23-53-17| photo_2023-11-21_23-54-11|
| photo_2023-11-21_23-54-56| photo_2023-11-21_23-55-15 |

### CODE REVIEW

**Comment by wielandb** [comment]
There is still one thing I need to address: I don't know how I access the tags of the object in question when creating the form, so that I can skip adding the "Other Answer" for the AddDisabledCapacity quest. Can someone point me in the right direction or show me where in the code something like that is already done?

**Comment by westnordost** [comment]
Was the third quest ever rejected upstream? On first glance, it looks potentially useful to me, but I don't have an overview over discussions / decisions already taken on that matter.

**Comment by Helium314** [comment]
Yes, it was rejected: https://github.com/streetcomplete/StreetComplete/issues/4163#issuecomment-1167270120

**Comment by westnordost** [comment]
Ah, I see, makes sense. Added one more comment in that ticket. Thanks for looking it up!

**Comment by wielandb** [comment]
> You seem to copy a lot of pretty huge vector graphics here. I would much prefer creating the orientation drawables like the [parking overlay does it](https://github.com/Helium314/SCEE/blob/393a2ec3d1fbc4d215f1adff8b1714e97ee3fd3e/app/src/main/java/de/westnordost/streetcomplete/overlays/street_parking/StreetParkingOverlayForm.kt#L163).

I have tried to create the images for the parking orientation quest dynamically to no avail now for some time. I simply cannot make dynamically created drawables play nicely in a **Quest where a selection is to be made**, and there are no quests where drawables are dynamically created in a simple "select one" context. I have now remade the form to be a simple text radio select, which I hope will suffice. If deemed insufficient, I can also remove this quest for now.

**Comment by westnordost** [comment]
Curious: What problem did you encounter? To the Android system, it shouldn't matter if one drawable is a vector drawable, a raster drawable or some completely custom drawable that draws itself dynamically.

**Comment by Helium314** [comment]
How do the capacity quest work exactly?
Should the normal capacity include disabled parking, because the question is "how many cars"?
Should the disabled capacity include normal parking? I'd say no, but I didn't check OSM wiki and wonder because you cannot answer 0.

> I have tried to create the images for the parking orientation quest dynamically to no avail now for some time

You can create the drawable using
`private fun ParkingOrientation.image(context: Context): DrawableImage = DrawableImage(StreetParkingDrawable(context, this, null, false, 128, 128, R.drawable.ic_car1))` (set the size or parking position to what you want).
Then switch to using `Item2`: `fun ParkingOrientation.asItem(context: Context) = Item2(this, image(context), ResText(titleResId))` and pass the context in the form.
Further you need to remove the duplication of `ParkingOrientation` and use the existing class instead.

This works for me, **but** the image disappears if it is selected.

**Comment by wielandb** [comment]
I would like to bring this PR to completion, as I still have a lot of ideas for PRs for SCEE that I would like to start working on. :) 

> This works for me, but the image disappears if it is selected. 

If I understand correctly, you also struggled getting this to work, so I would conclude that it is not (in whole) due to my lack of experience with Android development, but something that may be out of scope for now. 

So I see three options on how to continue regarding the `AddParkingOrientation` quest:
- Make the quest show the "big" images
- Make the quest a simple `AListQuestForm` (currently implemented in the PR)
- Remove the quest for now, and maybe revisit it later

How do you want to proceed @Helium314?

**Comment by wielandb** [comment]
I have a added hint about the capacity to `"quest_parking_capacity_hint"` as @mcliquid suggested. 

I have also made it possible to answer 0 for the disabled capacity quest.

**Comment by Helium314** [comment]
> If I understand correctly, you also struggled getting this to work, so I would conclude that it is not (in whole) due to my lack of experience with Android development, but something that may be out of scope for now.

I don't really remember, as that was quite a while ago... but I think getting the images to show was quite simple, and I didn't investigate further after noticing the issue when selecting the image (I think I was already working on something else when I noticed, and didn't bother to check further for the time being).

I think it's worth trying to use the `StreetParkingDrawable` as proposed above.

**Comment by wielandb** [comment]
> Curious: What problem did you encounter? To the Android system, it shouldn't matter if one drawable is a vector drawable, a raster drawable or some completely custom drawable that draws itself dynamically. But who knows, maybe there is a bug in the recycler view, or the recycler view never triggers a redraw on a drawable or something

I have tried once again and can now report where I'm stuck. 

In contrast to a simple resource, the function `StreetParkingDrawable()` needs a `context` as an argument (so that the images can be drawn in the right proportion to the screen, I think). That means, a `Context` object needs to be handed down through the function calls. I have looked at other Forms that use `Item2`, namely the `AddCyclewayQuest`, on how they handle it. When the call `asDialogItem()` (which is `asItem()` in my case) is made, `requireContext()` is passed as the `context` parameter.

https://github.com/Helium314/SCEE/blob/4e58ad719cc920798ce6482ee97ee8bb910204a5/app/src/main/java/de/westnordost/streetcomplete/quests/cycleway/AddCyclewayForm.kt#L171

So I tried doing the same thing:

```kotlin
    override val items = ParkingOrientation.values().map { it.asItem(requireContext()) }
```

It at least compiles fine, but as soon as the quest is opened, the following error occurs:

```
Process: de.westnordost.streetcomplete.expert.debug, PID: 25607
java.lang.IllegalStateException: Fragment AddParkingOrientationForm{9da85ca} (99166b38-8581-4c52-861f-d1dfe2df4391) not attached to a context.
at androidx.fragment.app.Fragment.requireContext(Fragment.java:972)
at de.westnordost.streetcomplete.quests.parking_orientation.AddParkingOrientationForm.<init>(AddParkingOrientationForm.kt:14)
at de.westnordost.streetcomplete.quests.parking_orientation.AddParkingOrientation.createForm(AddParkingOrientation.kt:24)
```
I am unsure how to "attach" this fragment to a context, as it works in `AddCyclewayForm` and I can't see anything special is done here to get a context. Help on this would be much apprechiated.

**Comment by Helium314** [comment]
The issue is that `items` are initialized before the `Fragment` has a `Context`

Instead of `override val items = ParkingOrientation.values().map { it.asItem(requireContext()) }` try `override val items get() = ParkingOrientation.values().map { it.asItem(requireContext()) }`.

**Comment by wielandb** [comment]
Thank you! That was the last piece to the puzzle!

Good news: It works! I managed to reproduce what @Helium314 mentioned was possible.
Bad news: Now I see the issue for myself... 😳

https://github.com/Helium314/SCEE/assets/29458682/85fa912c-a670-40ff-9c36-fa2e7ac8a440

> But who knows, maybe there is a bug in the recycler view, or the recycler view never triggers a redraw on a drawable or something

Pinging @westnordost, maybe you can make conclusions from this video if your theory is correct? (And maybe even know how to fix it if it's a quick fix? 😳)

**Comment by westnordost** [comment]
Sorry, no idea

**Comment by Helium314** [comment]
While I don't have an idea what's causing this, a simple solution is converting the drawable to a bitmap drawable (using `.asBitmapDrawable(context.resources)`.

@wielandb why did you choose to add new strings for parking orientation instead of using the existing ones? They are used for the same thing, and are already translated.

File: app/src/main/res/values/strings_ee.xml [hunk 1]
```diff
@@ -566,6 +566,19 @@
     <string name="quest_piste_difficulty_freeride">Freeride</string>
     <string name="quest_piste_difficulty_extreme">Extreme</string>
 
+    <!-- parking capacity quests -->
+    <string name="quest_parking_capacity_title">How many cars can park here?</string>
+    <string name="quest_parking_capacity_hint">This also includes special spaces like for disabled people, families, for charging etc. Hide the quest if the parking spaces are not marked and thus not countable</string>
+    <string name="quest_parking_capacity_disabled_title">How many parking spaces for disabled people are here?</string>
+    <string name="quest_parking_capacity_disabled_answer_yes">There are, but unclear count</string>
+    <string name="quest_parking_capacity_disabled_default_disabled_msg">Quests added in SCEE are always disabled by default, as they may not always be answerable, and may require knowledge of the topic and tagging.\nThis quest will not be asked if capacity:disabled=yes is set. If you want to specify them as well, change the element selection from "and !capacity:disabled" to "and (!capacity:disabled or capacity:disabled=yes)"
```
Comment (Helium314) [inline app/src/main/res/values/strings_ee.xml:574]: `"` is not visible in app, I think you need to escape it with `\`

Comment (wielandb) [reply app/src/main/res/values/strings_ee.xml]: Done

**Comment by wielandb** [comment]
> why did you choose to add new strings for parking orientation instead of using the existing ones? They are used for the same thing, and are already translated.

When the quest was a radio-select quest, I used strings with more explanation which were then boiled down to the one word descriptions. I removed them now and used the already available ones.

**Comment by Helium314** [comment]
Looks good now, only a few rather minor things I noticed:
* `showClarificationText` always has the same value for the respective form, so it can be removed together with `ARG_SHOW_CLARIFICATION´ and related code.
* `AddDisabledParkingCapacityForm` should better use `String` instead of `Int`, so the `-1` -> `yes` and `0` -> `no` can happen directly in the form.
* `AddParkingOrientationForm` has a bunch of unused imports
* `ParkingOrientation` could re-use the existing `ParkingOrientation` class and `ParkingOrientation.osmValue` (copy or make it public)
* `orientation_display_val` and `position_display_val` should follow the usual camelCase

**Comment by wielandb** [comment]
Implemented all your suggestions. Should be ready for review now. 🙂

**Pull request merged**